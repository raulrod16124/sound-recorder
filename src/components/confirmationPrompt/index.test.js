import { fireEvent, render, screen } from "@testing-library/react";

import ConfirmationPrompt from ".";

describe("Testing ConfirmationPrompt component on delete action", () => {
  let wrapper;

  //mocks
  const message = "Delete the recording Track001";
  const deleteRecording = jest.fn();
  const editRecordingName = jest.fn();
  const acceptButton = "Delete";
  const confirmationPromptState = {
    error: "",
  };
  const setConfirmationPromptState = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <ConfirmationPrompt
        message={message}
        acceptButton={acceptButton}
        confirmationPromptState={confirmationPromptState}
        deleteRecording={deleteRecording}
        editRecordingName={editRecordingName}
        setConfirmationPromptState={setConfirmationPromptState}
      />
    );
  });

  test("should render the ConfirmationPrompt component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should find the text Delete the recording Track001 in the document", () => {
    expect(
      screen.getByText("Delete the recording Track001")
    ).toBeInTheDocument();
  });

  test("should find the text cancel in the document", () => {
    expect(screen.getByText("cancel")).toBeInTheDocument();
  });

  test("should find the text Delete in the document", () => {
    expect(screen.getByText("Delete")).toBeInTheDocument();
  });
});

describe("Testing ConfirmationPrompt component on edit action", () => {
  let wrapper;

  //mocks
  const message = "Editing Track002";
  const deleteRecording = jest.fn();
  const editRecordingName = jest.fn();
  const acceptButton = "Update";
  const confirmationPromptState = {
    error: "The name is empty",
  };
  const setConfirmationPromptState = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <ConfirmationPrompt
        message={message}
        acceptButton={acceptButton}
        confirmationPromptState={confirmationPromptState}
        deleteRecording={deleteRecording}
        editRecordingName={editRecordingName}
        setConfirmationPromptState={setConfirmationPromptState}
      />
    );
  });

  test("should render the ConfirmationPrompt component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should find the text Editing Track002 in the document", () => {
    expect(screen.getByText("Editing Track002")).toBeInTheDocument();
  });

  test("should find the text New name in the document", () => {
    expect(screen.getByText("New name")).toBeInTheDocument();
  });

  test("should find textbox role", () => {
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  test("should find the text The name is empty in the document", () => {
    expect(screen.getByText("The name is empty")).toBeInTheDocument();
  });

  test("should find the text cancel in the document", () => {
    expect(screen.getByText("cancel")).toBeInTheDocument();
  });

  test("should find the text Update in the document", () => {
    expect(screen.getByText("Update")).toBeInTheDocument();
  });

  test("should call the setConfirmationPromptState onClick in the cancel button", () => {
    fireEvent.click(screen.getByText(/cancel/i));
    expect(setConfirmationPromptState).toHaveBeenCalledTimes(1);
  });
});
