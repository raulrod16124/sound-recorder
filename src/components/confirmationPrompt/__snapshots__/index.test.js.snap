// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmationPrompt component on delete action should render the ConfirmationPrompt component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="confirmation-prompt"
      >
        <p
          class="message"
        >
          Delete the recording Track001
        </p>
        <div
          class="content-buttons"
        >
          <button
            class="sc-bdvvtL czZeOQ"
            type="button"
          >
            cancel
          </button>
          <button
            class="sc-bdvvtL czZeOQ"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="confirmation-prompt"
    >
      <p
        class="message"
      >
        Delete the recording Track001
      </p>
      <div
        class="content-buttons"
      >
        <button
          class="sc-bdvvtL czZeOQ"
          type="button"
        >
          cancel
        </button>
        <button
          class="sc-bdvvtL czZeOQ"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing ConfirmationPrompt component on edit action should render the ConfirmationPrompt component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="confirmation-prompt"
      >
        <p
          class="message"
        >
          Editing Track002
        </p>
        <div
          class="content-input"
        >
          <label
            class="label-name"
          >
            New name
          </label>
          <input
            class="edit-input"
            placeholder="Write here the new name"
            type="text"
          />
        </div>
        <span
          class="error-message"
        >
          The name is empty
        </span>
        <div
          class="content-buttons"
        >
          <button
            class="sc-bdvvtL czZeOQ"
            type="button"
          >
            cancel
          </button>
          <button
            class="sc-bdvvtL czZeOQ"
            type="button"
          >
            Update
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="confirmation-prompt"
    >
      <p
        class="message"
      >
        Editing Track002
      </p>
      <div
        class="content-input"
      >
        <label
          class="label-name"
        >
          New name
        </label>
        <input
          class="edit-input"
          placeholder="Write here the new name"
          type="text"
        />
      </div>
      <span
        class="error-message"
      >
        The name is empty
      </span>
      <div
        class="content-buttons"
      >
        <button
          class="sc-bdvvtL czZeOQ"
          type="button"
        >
          cancel
        </button>
        <button
          class="sc-bdvvtL czZeOQ"
          type="button"
        >
          Update
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
